### ISSUE TITLE
@purpose should be able to be specified for <AudioLanguage> when used in <AccessibilityAttributes>

### ISSUE THREAD

**Issue by mitxp** [paulhiggs/dvb-i-tools#49]
Dear Paul,
thanks again for providing your DVB-I validator.
After using it for quite some time now, I happen to slightly disagree with the following validation code in lib/accessibility_attributes_checks.mjs:
```
        let checkLanguagePurpose = (elem, childName, errNum) => {
                elem?.forEachNamedChildElement(childName, (child) => {
                        // @purpose should not be specified for <AudioLanguage> when used in <AccessibilityAttributes>
                        if (child.attrAnyNs(tva.a_purpose))
                                errs.addError({
                                        code: `${errCode}-${errNum}1`,
                                        fragment: child,
                                        message: `${tva.a_purpose.attribute(child.name)} should not be specified for ${tva.e_AccessibilityAttributes.elementize()}`,
                                        key: keys.k_Accessibility,
                                        clause: "A177 table 56 (clause 6.10.10)",
                                });
                        checkAttributes(child, [], [], tvaEA.AudioLanguage, errs, `${errNum}2`);
                });
        };
```

You reference A177 table 56 (clause 6.10.10) as a reason for your check. When looking there, I read the following specification text:
_An item from the AudioPurposeCS vocabulary shall be assigned to the @purpose attribute of the element. See clause 6.11.3 for allowable CS terms. This element should not indicate the presence of audio description. See clause 4.5 for further details._

Now, when looking at 6.11.3, I only see an XPath  "InstanceDescription/AVAttributes/AudioAttributes/AudioLanguage[@purpose]". This is definitely not the case, as we are within the AccessibilityAttributes.

Now, let's look at chapter 4.5, or, to be more specific, at chapter 4.5.2.4: This chapter describes that audio description shall be indicated by the presence of one or more AccessibilityAttributes.AudioDescriptionAttributes elements. This is what I do.

One could argue that when placing the AudioAttributes.AudioLanguage inside AudioDescriptionAttributes, the purpose may not (instead of: should not) indicate the presence of audio description, so there should not be any purpose in this case. But certainly not for all purposes of AudioLanguage in AccessibilityAttributes.

However, I even think I should be allowed to use the AudioAttributes.AudioLanguage@purpose inside AudioDescriptionAttributes for two reasons:
- There is a big difference between urn:tva:metadata:cs:AudioPurposeCS:2007:1 (Audio description for visually impaired people) and urn:tva:metadata:cs:AudioPurposeCS:2007:2 (Audio description for hard of hearing people). I can only make that difference by adding the purpose to the AudioLanguage.
- DVB Bluebook A177 also does this in an example, see chapter 4.5.4.1 (AccessibilityAttributes in a ServiceList). In there, they give the following example:
```
<Service version="1">
  <UniqueIdentifier>tag:dvb.org,2023:advsid1</UniqueIdentifier>
  <!-- DVB-S2 Service -->
  <ServiceInstance priority="1">
    <ContentAttributes>
      <AudioAttributes>
        <tva:AudioLanguage purpose="urn:tva:metadata:cs:AudioPurposeCS:2007:6">en</tva:AudioLanguage>
      </AudioAttributes>
      <AudioAttributes>
        <tva:AudioLanguage purpose="urn:tva:metadata:cs:AudioPurposeCS:2007:6">es</tva:AudioLanguage>
      </AudioAttributes>
      <VideoConformancePoint href="urn:dvb:metadata:cs:VideoConformancePointsCS:2017:1.1.3"/>
      <AccessibilityAttributes>
        <tva:SubtitleAttributes>
          ...
        </tva:SubtitleAttributes>
        <tva:SubtitleAttributes>
          ...
        </tva:SubtitleAttributes>
        <tva:AudioDescriptionAttributes>
          <tva:AudioAttributes>
            <tva:Coding href="urn:dvb:metadata:cs:AudioCodecCS:2024:1.1.2"/>
             <tva:AudioLanguage purpose="urn:tva:metadata:cs:AudioPurposeCS:2007:1">en</tva:AudioLanguage>
          </tva:AudioAttributes>
        </tva:AudioDescriptionAttributes>
      </AccessibilityAttributes>
    </ContentAttributes>
    <DVBSDeliveryParameters>
       <DVBTriplet origNetId="1" serviceId="22222" tsId="3333"/>
    </DVBSDeliveryParameters>
  </ServiceInstance>
  ...
```

So in my point of view, the whole checkLanguagePurpose check should be removed. What do you think?

**Comment by paulhiggs** [comment]
Thank yo for supporting the use of the DVB-I validator and the comprehensive issue report. This change was made as a result of updated specification text for A177r8, which states the following for the `AudioLanguage` element in Table 56.

> The spoken language of the audio.
> When used in the description of the main programme audio, an item from the `AudioPurposeCS` vocabulary shall be assigned to the `@purpose` attribute of the element. See clause 6.11.3 for allowable CS terms.
> The `@purpose` attribute shall not be used when signalling the audio language of an accessibility feature, see clauses 4.5.2.4, 4.5.2.5 and 4.5.2.6.

My understanding from @andreastai is that this what was always expected when the `<AccessibilityAttributes>` element was introduced in TV-Anytime and A177. Our internal bugtracker has [issue 3293](https://bugzilla.dvb.org/show_bug.cgi?id=3293) for this has the following from Andreas Tai

> `From my perspective, the examples need to be corrected, and we should add a comment stating that the purpose attribute of the AudioAttributes element should not be used when describing audio-related accessibility features.`
> `Since the AudioAttributes element is already embedded in a specific Accessibility Element that indicates the type of accessibility feature, using the purpose value would simply duplicate this information and could lead to confusion.`
 
I will ask Andreas to look at this issue as well

**Comment by andreastai** [comment]
Thanks @mitxp for raising this. The issue is that otherwise we would have data redundancy and ambiguity when the purpose field is set differently compared to the type of accessibility attribute element where it is contained. Therefore it shall not be used. If a further differentation is need for audio related accessibility services this seems to be a new requirement and solved by another solution.

**Comment by mitxp** [comment]
Paul, many thanks for the clarification and your support. I only checked A177r6 and A177r7. Revision 8 explains it.
